src/Avalonia.X11/X11Screens.cs

```diff
@@ -82,6 +82,39 @@ private void OnEvent(XEvent ev)
                     _cache = null;
             }
 
+            private unsafe Size? GetPhysicalMonitorSizeFromEDID(IntPtr rrOutput)
+            {
+                if(rrOutput == IntPtr.Zero)
+                    return null;
+                var output = Marshal.ReadInt64(rrOutput);
+                var properties = XRRListOutputProperties(_x11.Display,output, out int propertyCount);
+                var hasEDID = false;
+                for(var pc = 0; pc < propertyCount; pc++)
+                {
+                    if(properties[pc] == _x11.Atoms.RR_PROPERTY_RANDR_EDID)
+                        hasEDID = true;
+                }
+                if(!hasEDID)
+                    return null;
+                XRRGetOutputProperty(_x11.Display, output, _x11.Atoms.RR_PROPERTY_RANDR_EDID, 0, 128, false, false, _x11.Atoms.AnyPropertyType, out int actualType, out int actualFormat, out int nItems, out _, out IntPtr prop);
```

Comment:
Where does 128 constant comes from?

Reply:
It's the length of EDID-Structure and it made sense in the code from chromium.
But after looking more into it i'm actually a bit confused.
An EDID can contain multiple extension blocks(each 128-Bytes), e.g Base + 2  Extensions => 384 Bytes.
The `length`-parameter is actually a `1/4` of the target size, so passing `128` would return the Base with 3 Extension.
We only need to pass `32` (base EDID), but i honestly don't get it.
Is the some documentation that can explain that behavior.

Reply:
According to https://cgit.freedesktop.org/xorg/proto/randrproto/tree/randrproto.txt, the offset and length are multyplied by 4.
Probably because XRRGetOutputProperty can return byte-,short-, or int-Arrays.